tests/components/modbus/test_modbus_sensor.py

```diff
@@ -491,6 +492,39 @@ async def test_config_wrong_struct_sensor(hass, do_config):
             [0x0102, 0x0304],
             str(int(0x04030201)),
         ),
+        (
+            {
+                CONF_COUNT: 1,
+                CONF_REGISTER_SIZE: 4,
+                CONF_DATA_TYPE: DATA_TYPE_CUSTOM,
+                CONF_STRUCTURE: ">HH",
+                CONF_SWAP: CONF_SWAP_BYTE,
+            },
+            [0x01020304],
+            "1027,513",
```

Comment:
It would be a lot easier to control if you have the result in hex.
What mean, you want 1 value returned.

however this is false you have count:1 and there are 2 values  returned,